### PR TITLE
Attribute Codex sessions to their model from thread settings

### PR DESCRIPTION
**Pull request by dalmasluca**
## Summary

Fixes the generic `codex` model bucket in `codex_usage_scanner.py` by attributing native Codex session tokens to their real model.

Two issues caused tokens to fall into a catch-all `codex` bucket:

1. Newer sessions record the model in `thread_settings_applied` (not `turn_context`), so the model was never picked up.
2. Token counts logged before the first model event were assigned to the default `codex` model, even when the session later revealed its model.

Both are fixed: the scanner now reads `thread_settings_applied.model`, and a pre-pass seeds `current_model` with the first model a session records.

Verified against real sessions: the `codex` bucket disappears and all tokens land on the actual models (e.g. `gpt-5.6-sol`, `gpt-5.6-terra`, `gpt-5.6-luna`, `gpt-5.4`...).

## Notes

- Authored by **opencode** (an AI coding assistant) on behalf of the reporter.
- This work was done by me (the assistant).

## Tests

```bash
./test/shell.d/model-usage-codex-scanner-test.sh
```

### CODE REVIEW

**Comment by dalmasluca** [comment]
Addressed the review comment plus a follow-up edge case (commit 47a7f0c7):

- pi sessions are now attributed to Codex strictly by `provider == "openai-codex"`, dropping the `api.startswith("openai-codex")` fallback. A third-party router connected in `pi agent` that serves the same gpt models — even one presenting an openai-codex-style api — no longer lands on the Codex screen.
- This matches the existing provider-only attribution for Kimi (`kimi-coding`) and Grok (`xai`).
- Regression test: a pi session from `third-party-router` with model `gpt-5.6-sol` and api `openai-codex-responses` is ignored, and leaves no `gpt-5.6-sol` bucket. All scanner tests pass.

File: bin/omarchy-agent-usage-codex [hunk 1]
```diff
@@ -194,6 +194,26 @@ def scan_native_codex_sessions():
 
   for path in files:
     current_model = "codex"
+    try:
+      # A session may log token counts before its first model event. Read the
+      # first model the file records so those early counts land on the real
+      # model instead of a generic "codex" bucket.
+      with path.open(errors="replace") as handle:
```
[the remark this answers is not in the document]
Comment (dalmasluca) [reply bin/omarchy-agent-usage-codex:260]: Resolved in acc72519. The new `rollout-thread.jsonl` fixture exercises exactly these two cases: a `token_count` precedes the first model event (line 49, so the pre-pass must skip token counts and keep scanning), and the only model event is an `event_msg` → `thread_settings_applied` with a direct `model` field (line 50, no `turn_context` at all). The asserts then check the early token count lands on `gpt-thread` and that no generic `codex` bucket is left behind. The same fixture drives the `turn_context` pre-pass path from the primary rollout, and the pi-router fixture asserts provider-strict filtering (47a7f0c7).

**Comment by dhh** [comment]
Heads up: #6603 just landed on `quattro` and restructured this whole area, so this PR needs a rework before it can merge.

The `model-usage` plugin is gone. It's been renamed to `omarchy.agents` (`shell/plugins/agents/`), and the panel is now strictly a display: it watches JSON records under `~/.local/state/omarchy/agents/usage/` and draws whatever appears there. All the in-plugin machinery this PR touches — `providers/*.qml` adapters and `scripts/*_usage_scanner.py` — no longer exists.

Usage collection now lives in standalone collectors in `bin/`. One `omarchy-agent-usage-<agent>` script per agent prints a complete, display-ready usage record (identity, tier, status, rate limits, today/week/all-time stats), and `omarchy-agent-usage-update` runs every collector it finds and writes the records atomically to the state directory.

The upside: adding an agent no longer touches the panel at all. Ship a collector that prints the record contract, optionally add an `assets/<id>.svg` mark (with an `-light` twin if it needs one for light surfaces), and the panel gains a tab on its own. See `shell/plugins/agents/README.md` for the contract and the `claude`/`codex` collectors in `bin/` as reference implementations — #6604 (Cursor) is a good example of a provider PR in the new style.

If you'd like to update this PR to the new structure, rebase on `quattro` and reshape the scanner into a collector. Happy to answer questions here.

— Claude Fable 5, written on behalf of @dhh
